File: web-basics/WebLab3/src/main/java/com/arslanka/backend/services/HitCheckService.java
package com.arslanka.backend.services;

import com.arslanka.backend.models.daos.Request;
import com.arslanka.backend.models.views.Circle;
import com.arslanka.backend.models.views.Distance;
import com.arslanka.backend.models.views.Graph;
import com.arslanka.backend.models.views.Rectangle;
import com.arslanka.backend.models.views.Triangle;
import jakarta.enterprise.context.ApplicationScoped;
import jakarta.inject.Named;

@Named("hitCheckService")
@ApplicationScoped
public class HitCheckService {

    private final Graph graph;


    {
        graph = new Graph(
                null,
                new Rectangle(Distance.R, Distance.HALF_R),
                new Triangle(Distance.R, Distance.R),
                new Circle(Distance.HALF_R));
    }

    public Boolean checkStatus(Request request) {
        return isHit(request);
    }

    private Boolean isHit(Request request) {
        return graph.isHit(request.getRadius(), request.getPoint2D().getX(), request.getPoint2D().getY());
    }
}

File: web-basics/WebLab4/src/main/kotlin/com/example/arslanka/weblab4/services/HitCheckService.kt
package com.example.arslanka.weblab4.services

import com.example.arslanka.weblab4.models.Point2D
import com.example.arslanka.weblab4.models.views.Graph
import org.springframework.stereotype.Service
import java.math.BigDecimal

@Service
class HitCheckService(private val graph: Graph) : HitChecker {
    override fun checkStatus(point2D: Point2D, radius: BigDecimal): Boolean {
        return isHit(x = point2D.x, y = point2D.y, radius = radius)
    }

    private fun isHit(x: BigDecimal, y: BigDecimal, radius: BigDecimal): Boolean {
        return graph.isHit(x = x, y = y, radius = radius)
    }
}